processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -27,45 +28,99 @@ import (
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-func TestJobE2E(t *testing.T) {
-	f, err := os.Open("./testdata/trace.json")
-	assert.NoError(t, err)
+func TestTraceE2E(t *testing.T) {
+
+	tcs := []struct {
+		name  string
+		attrs map[string]string
+	}{
+		{
+			name: "job",
+			attrs: map[string]string{
+				"k8s.namespace.name": "default",
+				"k8s.job.name":       "test-telemetrygen-job",
+			},
+		},
+		{
+			name: "statefulset",
+			attrs: map[string]string{
+				"k8s.namespace.name":   "default",
+				"k8s.statefulset.name": "test-telemetrygen-statefulset",
+			},
+		},
+		{
+			name: "deployment",
+			attrs: map[string]string{
+				"k8s.namespace.name":  "default",
+				"k8s.deployment.name": "test-telemetrygen-deployment",
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := scanSpanForAttributes(tc.attrs)
+			assert.NoError(t, err)
+			assert.Equal(t, true, result)
+		})
+	}
+}
+
+func scanSpanForAttributes(kvs map[string]string) (bool, error) {
+	f, _ := os.Open("./testdata/trace.json")
 	defer f.Close()
 
 	scanner := bufio.NewScanner(f)
-
 	const maxCapacity int = 8388608
 	buf := make([]byte, maxCapacity)
 	scanner.Buffer(buf, maxCapacity)
 
 	unmarshaler := ptrace.JSONUnmarshaler{}
-	jobKeyFound := false
-	nsKeyFound := false
 	for scanner.Scan() {
 		traces, err := unmarshaler.UnmarshalTraces(scanner.Bytes())
-		assert.NoError(t, err)
+		if err != nil {
+			return false, err
+		}
 
 		len := traces.ResourceSpans().Len()
 		for i := 0; i < len; i++ {
 			resourceSpans := traces.ResourceSpans().At(i)
-
-			resourceSpans.Resource().Attributes().Range(
-				func(k string, v pcommon.Value) bool {
-					if k == "k8s.job.name" && v.Str() == "test-telemetrygen" {
-						jobKeyFound = true
-					}
-					if k == "k8s.namespace.name" && v.Str() == "default" {
-						nsKeyFound = true
-					}
-					return true
-				},
-			)
+			if resourceHasAttributes(resourceSpans.Resource(), kvs) {
+				return true, err
+			}
 		}
 	}
 
-	assert.True(t, jobKeyFound && nsKeyFound, "job and namespace attributes are not found")
-
 	if err := scanner.Err(); err != nil {
-		assert.NoError(t, err)
+		return false, err
 	}
+
+	return false, fmt.Errorf("attributes are not in span")
+}
+
+func resourceHasAttributes(resource pcommon.Resource, kvs map[string]string) bool {
+	exists := make(map[string]bool)
+	for k := range kvs {
+		exists[k] = false
+	}
+
+	resource.Attributes().Range(
+		func(k string, v pcommon.Value) bool {
+			if val, ok := kvs[k]; ok {
+				if val == v.AsString() {
+					exists[k] = true
+				}
+			}
+			return true
+		},
+	)
+
+	has := true
+	for _, v := range exists {
+		if !v {
+			has = false
```

Comment:
```suggestion
			return false
```
nit, should be the same ^^